source/_components/switch.arest.markdown

```diff
@@ -39,18 +42,39 @@ switch:
       function1:
         name: Light Desk
 ```
-
-Configuration variables:
-
-- **resource** (*Required*): IP address and schema of the device that is exposing an aREST API, eg. http://192.168.1.10 (no-trailing slash)
-- **name** (*Optional*): Let you overwrite the name of the device. By default *name* from the device is used.
-- **pins** array (*Optional*): An array with all used pins.
-  - **name** (*Required*): The name of the pin to use in the frontend.
-
-or 
-
-- **functions** array (*Optional*): An array with all used functions.
-  - **name** (*Required*): The name to use in the frontend.
+{% configuration %}
+resource:
+  description: IP address and schema of the device that is exposing an aREST API, eg. http://192.168.1.10 (no-trailing slash)
+  required: true
+  type: string
+name:
+  description: Let you overwrite the name of the device. By default *name* from the device is used.
+  required: optional
+  type: string
+pins:
+  description: An array with all used pins.
+  required: optional
+  type: map
+  keys:
+    name:
+      description: The name of the pin to use in the frontend.
+      required: true
+      type: string
+    invert:
+      description: If the logic for on/off should be inverted.
+      required: optional
```

Comment:
`required: false`. Is `optional` also valid?

Reply:
Well valid as in "Does it work?"; yes. Matching our standards: no.